doc/devices/dts/device_tree.rst

```diff
@@ -80,6 +80,54 @@ This .fixup file by default resides in the board directory and is named
 dts.fixup.  This fixup file maps the generated include information to the
 current driver/source usage.
 
+.. _dt_vs_kconfig:
+
+Device Tree vs Kconfig
+**********************
+
+As mentioned above there are several tools used to configuring the build with
+Zephyr.
+The two main ones, Device Tree and Kconfig might seem to overlap in purpose,
+but in fact they do not. This section is intended to serve as a reference when
+deciding whether to place a certain configuration item in Device Tree or Kconfig.
+
+The scope of each configuration tool can be summarized as follows:
+
+* Device Tree is used to describe the **hardware** and its **boot-time
+  configuration**.
+* Kconfig is used to describe which **software features** will be built into
+  the final image, and their **configuration**.
```

Comment:
firmware features?

Reply:
I chose to phrase this differently, see last version of the commit